workflows/diagnostics/tests/offline/test_helpers.py

```diff
@@ -112,12 +114,28 @@ def test_insert_column_integrated_vars():
     xr.testing.assert_allclose(insert_column_integrated_vars(ds, ["Q1"]), expected)
 
 
+def test_load_grid_info_from_custom_catalog(datadir):
+    # This test is dedicated to validate the response of "load_grid_info" function
+    # on a bogus catalog file. Retrieving the grid information with a non-existing
+    # resolution string generates the "KeyError".
+
+    res = "c12"
+    catalog_path = os.path.join(datadir, "catalog_dummy.yaml")
+
+    with pytest.raises(KeyError):
+        load_grid_info(res, catalog_path)
+
+
 @pytest.mark.parametrize(
-    "res, dim_name, expected_size",
-    [("c12", "x", 12), ("c48", "x", 48), ("ne30", "ncol", 21600),],
+    "res, catalog_path, dim_name, expected_size",
+    [
+        ("c12", vcm.catalog.catalog_path, "x", 12),
+        ("c48", vcm.catalog.catalog_path, "x", 48),
+        ("ne30", vcm.catalog.catalog_path, "ncol", 21600),
+    ],
 )
-def test_load_grid_info(res, dim_name, expected_size):
-    grid = load_grid_info(res)
+def test_load_grid_info(res, catalog_path, dim_name, expected_size):
+    grid = load_grid_info(res, catalog_path)
```

Comment:
This test needs to be updated based on the new signature of `load_grid_info`:
```suggestion
def test_load_grid_info(res, catalog, dim_name, expected_size):
    grid = load_grid_info(catalog, res)
```